### PR TITLE
feat(mev-build): support multiple forks when submitting bid

### PR DESCRIPTION
**Pull request by chirag-bgh**
- Implemented retrieving `Execution Payload` for `Bellatrix` & `Deneb` forks

### CODE REVIEW

File: mev-build-rs/src/reth_builder/reth_compat.rs [hunk 1]
```diff
@@ -58,7 +113,54 @@ pub(crate) fn to_execution_payload(value: &SealedBlock) -> ExecutionPayload {
...
+
+    let blob_tx_hashes = value.blob_transactions().iter().map(|t| t.hash()).collect::<Vec<_>>();
+    let blobs_bundle = get_blob_bundles(pool, blob_tx_hashes);
+    let deneb_payload_bundles =
+        deneb::ExecutionPayloadAndBlobsBundle { execution_payload: deneb_payload, blobs_bundle };
+
+    ExecutionPayload::Deneb(deneb_payload_bundles)
+}
+
+fn get_blob_bundles<Pool>(pool: &Pool, tx_hashes: Vec<H256>) -> BlobsBundle
```
Comment (ralexstokes) [inline mev-build-rs/src/reth_builder/reth_compat.rs:129]: this logic would be encapsulated in the `PayloadBuilder` abstraction so we wouldn't need to pass this `Pool` generic around through all of this code

File: mev-build-rs/src/reth_builder/reth_compat.rs [hunk 2]
```diff
@@ -22,7 +30,10 @@ pub(crate) fn to_u256(value: &U256) -> ssz_rs::U256 {
     ssz_rs::U256::try_from_bytes_le(&value.to_le_bytes::<32>()).unwrap()
 }
 
-pub(crate) fn to_execution_payload(value: &SealedBlock) -> ExecutionPayload {
+pub(crate) fn to_execution_payload<Pool>(value: &SealedBlock, pool: &Pool) -> ExecutionPayload
```
Comment (ralexstokes) [inline mev-build-rs/src/reth_builder/reth_compat.rs:33]: it seems better to just pass in the fork schedule (perhaps just mapping from timestamp to fork) and then dispatch to the relevant type?

File: mev-build-rs/src/reth_builder/builder.rs [hunk 3]
```diff
@@ -292,7 +297,7 @@ impl<Pool, Client> Builder<Pool, Client> {
             let block_hash = &signed_submission.message.block_hash;
             let value = &signed_submission.message.value;
             tracing::info!(id = %id, relay = ?relay, slot, %parent_hash, %block_hash, ?value, %builder_payment, "submitting bid");
-            match relay.submit_bid(&signed_submission).await {
+            match relay.submit_bid(&signed_submission.clone()).await {
```
Comment (ralexstokes) [inline mev-build-rs/src/reth_builder/builder.rs:300]: why clone here?

**Review by ralexstokes — COMMENTED** [review]
thanks for this!

we do want to do something here but let's consider the design a bit first

do you know how reth handles polymorphism of the `SealedBlock` type? is it just `Option<T>` for the relevant `T`s for each fork?

File: mev-build-rs/src/reth_builder/reth_compat.rs (hunk 2 above)
Comment (chirag-bgh) [reply mev-build-rs/src/reth_builder/reth_compat.rs]: I fixed this by fetching the currnet_epoch and comparing with the differents fork's epochs

**Comment by chirag-bgh** [comment]
> do you know how reth handles polymorphism of the `SealedBlock` type? is it just `Option<T>` for the relevant `T`s for each fork?

I may not be aware of this but I think there's just one form for `SealedBlock` which handles all the forks

**Pull request closed without merging**